### PR TITLE
more homogenous no-unit testing

### PR DESCRIPTION
**Pull request by volodymyrss**
Maybe a bit shorter. Mostly that more homogenous treatment between `int` and `float` and more clear table of what maps to what.

For more complex cases like with units this approach might not be better, not sure.

### CODE REVIEW

File: cdci_data_analysis/analysis/parameters.py [hunk 1]
```diff
@@ -408,7 +408,7 @@ def check_float_value(value, units=None, name=None):
             try:
                 value = float(value)
             except:
-                raise RuntimeError('type %s not valid for %s' % (type(value), name))
+                raise RuntimeError(f'unable to interpret value {value} (of type {type(value)}) as float')
```
Comment (burnout87) [inline cdci_data_analysis/analysis/parameters.py:411]: funny enough, I had applied this change also in the other PR #280 , before checking this

**Review by burnout87 — COMMENTED** [review]
ok it helps in having two tests in one, but perhaps I would keep them separate to better inspect specific behaviors of the two classes (`Float`, `Integer`), also in case we apply changes in the future

File: cdci_data_analysis/analysis/parameters.py (hunk 1 above)
Comment (volodymyrss) [reply cdci_data_analysis/analysis/parameters.py:411]: this is the merge on `parameters-default-format`, hence any visible change should not be yet in `parameters-default-format` - in #280 . If it is, rebase before merge will hide it

**Comment by volodymyrss** [comment]
> ok it helps in having two tests in one, but perhaps I would keep them separate to better inspect specific behaviors of the two classes (`Float`, `Integer`), also in case we apply changes in the future

much of the behavior between `Float` and `Integer` is almost identical. In fact, whatever remains different, does not appear to me justified. This is why, to simplify dicussion, I suggest this.

However, if there is any particular difference between `Float` and `Integer` which is not covered by the test I suggest, please let me know, I could have missed it!

File: cdci_data_analysis/analysis/parameters.py (hunk 1 above)
Comment (volodymyrss) [reply cdci_data_analysis/analysis/parameters.py:414]: I indeed appeared as a conflict and I resolved it - by picking a slightly longer and more detailed message. Do you have better suggestions for the text?

**Review by burnout87 — APPROVED** [review]
Just one suggestion for the text in error message, which should also applied to the Integer class

**Pull request merged**